src/Physics/HadronTransport/INukeHadroData2025.cxx

```diff
@@ -0,0 +1,1900 @@
+    else if ( fate == kIHAFtInelas ) return frac_inelas / total;
+    else if ( fate == kIHAFtAbs    ) return frac_abs / total;
+    else if ( fate == kIHAFtPiProd ) return frac_pipro / total;
+    else if ( fate == kIHAFtCmp    ) return frac_comp / total; // cmp - add support for this later
+    else {
+      LOG("INukeData", pWARN)
+        << "Protons don't have this fate: " << INukeHadroFates::AsString(fate);
+      return 0;
+    }
+  }
+  else if (hpdgc == kPdgNeutron) {
+    // handle neutrons
+    double frac_cex = fFracNA_CEx->Evaluate(ke);
+    double frac_inelas = fFracNA_Inel->Evaluate(ke);
+    double frac_abs = fFracNA_Abs->Evaluate(ke);
+    double frac_pipro = fFracNA_PiPro->Evaluate(ke);
+    double frac_comp = fFracNA_Cmp->Evaluate(ke);
+
+    // Protect against unitarity violation due to interpolation problems
+    // by renormalizing all available fate fractions to unity.
+    double total = frac_cex + frac_inelas + frac_abs + frac_pipro + frac_comp; // + frac_elas
+
+    if ( fate == kIHAFtCEx ) return frac_cex / total;
+  //else if ( fate == kIHAFtElas   ) return frac_elas / total;
+    else if ( fate == kIHAFtInelas ) return frac_inelas / total;
+    else if ( fate == kIHAFtAbs    ) return frac_abs / total;
+    else if ( fate == kIHAFtPiProd ) return frac_pipro / total;
+    else if ( fate == kIHAFtCmp    ) return frac_comp / total; // cmp - add support for this later
+    else {
+      LOG("INukeData", pWARN)
+        << "Neutrons don't have this fate: " << INukeHadroFates::AsString(fate);
+      return 0;
+    }
+  }
+  else if (hpdgc == kPdgKP) {
+    // handle K+
+    double frac_inelas = fFracKA_Inel->Evaluate(ke);
+  //double frac_elas = fFracKA_Elas->Evaluate(ke);
+    double frac_abs = fFracKA_Abs->Evaluate(ke);
+
+    // Protect against unitarity violation due to interpolation problems
+    // by renormalizing all available fate fractions to unity.
+    double total = frac_inelas + frac_abs; // + frac_elas
+
+    if ( fate == kIHAFtInelas ) return frac_inelas / total;
+    else if ( fate == kIHAFtAbs ) return frac_abs / total;
+    else {
+      LOG("INukeData", pWARN)
+        << "K+'s don't have this fate: " << INukeHadroFates::AsString(fate);
+      return 0.;
+    }
+  }
+  LOG("INukeData", pWARN) << "Can't handle particles with pdg code = " << hpdgc;
+  return 0.;
+}
+//____________________________________________________________________________
+double INukeHadroData2025::XSec(int hpdgc, INukeFateHN_t fate, double ke, int targA, int targZ) const
+{
+// return the x-section for the input fate for the particle with the input pdg
+// code at the input kinetic energy
+//
+  ke = TMath::Max(fMinKinEnergy,   ke);
+  ke = TMath::Min(fMaxKinEnergyHN, ke);
+
+  LOG("INukeData", pDEBUG)  << "Querying hN cross section at ke = " << ke;
+
+  double xsec=0;
+
+    if (hpdgc == kPdgPiP) {
+    /* handle pi+ */
+         if (fate == kIHNFtCEx   ) {xsec = TMath::Max(0., fXSecPipp_CEx  -> Evaluate(ke)) *  targZ;
```

Comment:
Formatting these as 
<pre>
  if (fate == blah ) {
     xsec  = ...;
     xsec += ...;
    return xsec
  }</pre>. all lined up, or as:
<pre>
  if (fate == blah) {
     return TMath::Max() * targZ +
            TMath::Max() * (targA-targZ);
  }</pre> might make things easier to read and possibly spot potential typos